kratos/python_scripts/restart_utility.py

```diff
@@ -0,0 +1,160 @@
+            "serializer_trace"              : "no_trace",
+            "echo_level"                    : 0
+        }
+        """)
+
+        __serializer_flags = {
+            "no_trace":           KratosMultiphysics.SerializerTraceType.SERIALIZER_NO_TRACE,     # binary
+            "trace_error":        KratosMultiphysics.SerializerTraceType.SERIALIZER_TRACE_ERROR,  # ascii
+            "trace_all":          KratosMultiphysics.SerializerTraceType.SERIALIZER_TRACE_ALL     # ascii
+        }
+
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        self.model_part = model_part
+
+        # load settings
+        self.input_filename = settings["input_filename"].GetString()
+        self.input_file_label = settings["restart_load_file_label"].GetString()
+
+        # save settings
+        self.restart_save_frequency = settings["restart_save_frequency"].GetDouble()
+        serializer_trace = settings["serializer_trace"].GetString()
+
+        restart_control_type = settings["restart_control_type"].GetString()
+        if restart_control_type == "time":
+            self.restart_control_type_is_time = True
+        elif restart_control_type == "step":
+            self.restart_control_type_is_time = False
+        else:
+            err_msg =  'The requested restart_control_type "' + restart_control_type + '" is not available!\n'
+            err_msg += 'Available options are: "time", "step"'
+            raise Exception(err_msg)
+
+        serializer_trace = settings["serializer_trace"].GetString()
+        if not serializer_trace in __serializer_flags.keys():
+            err_msg =  'The requested serializer_trace "' + serializer_trace + '" is not available!\n'
+            err_msg += 'Available options are: "no_trace", "trace_error", "trace_all"'
+            raise Exception(err_msg)
+        self.serializer_flag = __serializer_flags[serializer_trace]
+
+        self.next_output = 0.0
+
+        self.echo_level = settings["echo_level"].GetDouble()
+
+        self.move_restart_files_to_folder = settings["move_restart_files_to_folder"].GetBool()
+
+    #### Public functions ####
+
+    def LoadRestart(self):
+        """
+        This function loads a restart file into a ModelPart
+        """
+        # Get file name
+        restart_path = self.__GetFileNameLoad()
+        # Check path
+        if (os.path.exists(restart_path+".rest") == False):
+            raise Exception("Restart file not found: " + restart_path + ".rest")
+        self._PrintOnRankZero("::[Restart Utility]::", "Loading restart file:", restart_path + ".rest")
+
+        # Load the ModelPart
+        serializer = KratosMultiphysics.Serializer(restart_path, self.serializer_flag)
+        serializer.Load(self.model_part.Name, self.model_part)
+
+        self._ExecuteAfterLaod()
+
+        self.model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED] = True
+        load_step = self.model_part.ProcessInfo[KratosMultiphysics.STEP] + 1
```

Comment:
Are you sure you want to increase the counter here? From the point of view of the solution, you will have the step counter 1 higher for each restart you do in between. It is a very minor point but it could have some impact if you use STEP to, for example, compute time averages.